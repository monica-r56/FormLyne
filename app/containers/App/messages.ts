import { defineMessages } from 'react-intl';

export default defineMessages({
  root_app: {
    id: 'Root.root_app',
    defaultMessage: 'Root App',
  },
  something_went_wrong: {
    id: 'Root.something_went_wrong',
    defaultMessage: 'Something went wrong',
  },
});
