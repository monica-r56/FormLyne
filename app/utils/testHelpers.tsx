import React, { type ReactNode } from 'react';
import { type RenderOptions, RenderResult, render as rtlRender } from '@testing-library/react';
import { LanguageProvider } from '@contexts/LanguageContext';
import { MemoryRouter } from 'react-router';

interface ExtendedRenderOptions extends RenderOptions {
  initialPath?: string;
  route?: string[];
}

class ResizeObserver {
  observe() {}

  unobserve() {}

  disconnect() {}
}

window.ResizeObserver = ResizeObserver;

function render(ui: React.ReactNode, options?: ExtendedRenderOptions): RenderResult {
  const Wrapper = (props: { children: ReactNode }) => (
    <MemoryRouter initialEntries={options?.route || ['/']}>
      <LanguageProvider>{props.children}</LanguageProvider>
    </MemoryRouter>
  );

  return { ...rtlRender(ui, { wrapper: Wrapper, ...options }) };
}

export * from '@testing-library/react';

export { render };
