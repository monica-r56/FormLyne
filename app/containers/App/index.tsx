import { FormattedMessage } from 'react-intl';
import messages from './messages';

const App = () => (
  <div className={'flex h-screen w-screen flex-1 items-center justify-center'}>
    <FormattedMessage {...messages.root_app} />
  </div>
);

export default App;
