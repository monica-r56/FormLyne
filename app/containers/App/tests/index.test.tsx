// import { render } from '@testing-library/react';
import { render } from '@utils/testHelpers';
import App from '..';

describe('App Component', () => {
  it.skip('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(<App />);
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });
});
