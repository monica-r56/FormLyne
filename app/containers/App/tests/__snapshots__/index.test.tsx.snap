// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Should render and match the snapshot 1`] = `
<div>
  <div
    class="flex h-screen w-screen flex-1 items-center justify-center"
  >
    Root App
  </div>
</div>
`;
